include/erlcloud_aws.hrl

```diff
@@ -42,7 +42,7 @@
           secret_access_key::string()|undefined|false,
           security_token=undefined::string()|undefined,
           %% epoch seconds when temporary credentials will expire
-          expiration :: pos_integer(),
+          expiration=undefined :: pos_integer()|undefined,
```

Comment:
It's not necessary; I did it because it's in a block of others that all have an explicit default value and `security_token` right above this already had an explicit-undefined default.

I think that most of the values that get an implicit `undefined` do that from laziness, but some actually mean it. For example, all the response-related values below this in the record are definitely `undefined` until the response comes back and they get filled in.

My preference would be to make defaults explicit, because then `x=undefined::integer()|undefined` means "this can actually be undefined (because it's filled in conditionally, or there's a special meaning when it's not defined)" and `x::integer()|undefined` means "this is lazy-code and we don't actually know if `x` needs to accept a value of `undefined`.

Odds of someone carefully refactoring the whole code-base to eliminate unnecessary implicit `undefined`s are real low, though.